arangod/Agency/Supervision.cpp

```diff
@@ -860,13 +861,139 @@ void Supervision::handleShutdown() {
   }
 }
 
+void Supervision::cleanupLostCollections(Node const& snapshot, AgentInterface *agent, std::string const& jobId) {
+
+  std::unordered_set<std::string> failedServers;
+
+  LOG_TOPIC(ERR, Logger::FIXME) << "Snapshot: " << snapshot;
+
+  // Search for failed server
+  //  Could also use `Target/FailedServers`
+  auto const& health = snapshot.hasAsChildren(healthPrefix);
+  if (!health.second) {
+    return ;
+  }
+
+  LOG_TOPIC(ERR, Logger::FIXME) << "called cleanupLostCollections()";
+
+  for (auto const& server : health.first) {
+    HealthRecord record(*server.second.get());
+
+    if (record.status == Supervision::HEALTH_STATUS_FAILED) {
+      failedServers.insert(server.first);
+      LOG_TOPIC(ERR, Logger::FIXME) << "Found failed server: " << server.first;
+    }
+  }
+
```

Comment:
According to spec you wanted to break out here if the list of failed servers is empty.

Reply:
yes. what about that?